### PR TITLE
📊 world happiness report

### PR DESCRIPTION
**Pull request by antea04**
See also: owid/owid-issues#1524
Update Happiness data with 2024 world happiness report and add aggregates over continent/ age group

Goals:
- [x]  Check that the methodology of the WHR is consistent across years, and it is okay for us to combine data across multiple reports, as we currently do
  - [x] If that is okay combine the new data with the current data in the etl
- [x] Calculate population-weighted continent and income group averages and potentially a global average of the Cantril Ladder Scores
- (data for this currently not available - won't do) Import the Cantril Ladder Scores for the different age-groups and draft a chart showing the differences across age-groups

Low-level tasks
- [x] Snapshot/ import WHR 2024 data
- [x] Read methodology and whether the data is comparable over years
- [x] Figure out whether data for age group is available (reached out - no answer yet)
- [x] Continent/ income group averages calculation (garden step)
- [x] Update existing charts (check charts by indicator)
  - (data for this not available) New chart (comp over age groups)

### CODE REVIEW

File: etl/steps/data/garden/happiness/2024-06-09/happiness.meta.yml
Comment (spoonerf) [inline etl/steps/data/garden/happiness/2024-06-09/happiness.meta.yml:47]: Maybe this section of hashed out config can be deleted

File: etl/steps/data/garden/happiness/2024-06-09/happiness.py
Comment (spoonerf) [inline etl/steps/data/garden/happiness/2024-06-09/happiness.py:12]: Fancy!

File: etl/steps/data/garden/happiness/2024-06-09/happiness.py
Comment (spoonerf) [inline etl/steps/data/garden/happiness/2024-06-09/happiness.py:48]: You could consider excluding these countries, but I'm also happy if you want to keep them in.

Comment (spoonerf) [reply etl/steps/data/garden/happiness/2024-06-09/happiness.py:48]: Also perhaps a slightly neater way to achieve the same thing would be:

```
msk = tb["country"].isin(["Northern Cyprus", "Somaliland"])
tb_countries_wo_population = tb[msk]
tb = tb[~msk]
```

**Review by spoonerf — APPROVED** [review]
Looks good to me! Just some minor comments on my side, but I would wait for Pablo R's review before merging. 

I'm not sure why the Marimekko chart isn't showing as expected on chart-diff, perhaps you could report this in #proj-new-data-workflow. 
Screenshot 2024-06-14 at 11 16 01

File: etl/steps/data/garden/happiness/2024-06-09/happiness.meta.yml
Comment (spoonerf) [inline etl/steps/data/garden/happiness/2024-06-09/happiness.meta.yml:16]: If possible, it would be good to retain the links to the previous reports ([like here](https://github.com/owid/etl/blob/41f9b1d7f40fdf19fd8270cb79d03237d5e6708a/etl/steps/data/garden/happiness/2023-03-20/happiness.meta.yml#L18)), as this dataset is a combination of many annual. I'm not sure what would be the best place to do this, though.

This information is stored in each of the Snapshots of the source files, but these do not use the Origin format. It could be a bit of a headache to fix, I'd be curious to hear what @pabloarosado thinks about it.

File: etl/steps/data/garden/happiness/2024-06-09/happiness.py
Comment (antea04) [reply etl/steps/data/garden/happiness/2024-06-09/happiness.py:48]: I would keep them in - they only show up if they get selected in the line chart and there I think it might be nice to have them available.

File: snapshots/happiness/2024-06-09/happiness.xls.dvc
Comment (antea04) [reply snapshots/happiness/2024-06-09/happiness.xls.dvc:25]: I did struggle with that because on the website there is no license mentioned or explicit instructions for how to (re-)publish the data. THe closest thing would be the citation and isbn on the report page (https://worldhappiness.report/ed/2024/). I added this now as license url - let me know if that works also!

File: snapshots/happiness/2024-06-09/happiness.xls.dvc
Comment (pabloarosado) [inline snapshots/happiness/2024-06-09/happiness.xls.dvc:14]: This could be simply "Wellbeing Research Centre". That's how they call themselves in multiple public places. And you mention Oxford in the full citation, so that should be fine.

File: etl/steps/data/meadow/happiness/2024-06-09/happiness.py
Comment (pabloarosado) [inline etl/steps/data/meadow/happiness/2024-06-09/happiness.py:22]: Instead you could simply do `tb = tb.rename(columns={"Country name": "country"}, errors="raise")`.

File: etl/steps/data/grapher/happiness/2024-06-09/happiness.py
Comment (pabloarosado) [inline etl/steps/data/grapher/happiness/2024-06-09/happiness.py:21]: If there is no processing, you can remove the "Process data" section.

File: etl/steps/data/garden/happiness/2024-06-09/happiness.py
Comment (pabloarosado) [inline etl/steps/data/garden/happiness/2024-06-09/happiness.py:71]: Given that population was crucial for this calculation, you could leave it as part of the data. Imagine that in some months from now you discover some issue in one of the regions. Having population there would help to find out what happened (yes, we also have the population dataset, so it's not totally necessary).

File: etl/steps/data/garden/happiness/2024-06-09/happiness.meta.yml
Comment (pabloarosado) [reply etl/steps/data/garden/happiness/2024-06-09/happiness.meta.yml:16]: Are we expecting to keep using previous versions of the report every year? In that case, maybe it's worth moving the old step to the new metadata format. If the snapshot doesn't exist anymore, we could just change the snapshot dvc file, and the garden meta yml file (and possibly some minor things in the code).

**Review by pabloarosado — APPROVED** [review]
Looks good! I suggested that maybe it's worth adapting the step of the previous year to the new metadata format, if we are planning on re-using it in the future.

File: etl/steps/data/garden/happiness/2024-06-09/happiness.meta.yml
Comment (antea04) [reply etl/steps/data/garden/happiness/2024-06-09/happiness.meta.yml:16]: Thanks for the review @pabloarosado - indeed I think it would be good to move the old data set to origins as well. I will follow the steps you laid out - but what do you mean by "if the snapshot doesn't exist anymore"? the old snapshots from 2012-2023 are still under etl/snapshots/happiness

Comment (pabloarosado) [reply etl/steps/data/garden/happiness/2024-06-09/happiness.meta.yml:16]: Good idea! What I meant is that the data provider's data may not be available anymore. If that's the case, you will not be able to execute the snapshot .py file. However, even then, you can simply modify the content of the .dvc file, and ETL will read the new metadata as expected. There's no need to re-execute the corresponding .py file. Does that make sense?

**Comment by pabloarosado** [comment]
> Looks good to me! Just some minor comments on my side, but I would wait for Pablo R's review before merging.
> 
> I'm not sure why the Marimekko chart isn't showing as expected on chart-diff, perhaps you could report this in #proj-new-data-workflow. Screenshot 2024-06-14 at 11 16 01

[That chart seems to be a draft chart in production](https://admin.owid.io/admin/variables/577009), and it's not working in production either, so I would simply ignore it (or even consider removing it).

**Comment by spoonerf** [comment]
> [That chart seems to be a draft chart in production](https://admin.owid.io/admin/variables/577009), and it's not working in production either, so I would simply ignore it (or even consider removing it).

Good point! I just deleted it.

**Pull request merged**
